src/rc.c

```diff
@@ -178,7 +186,27 @@ read_rc(const char *file)
 		abs_file = NULL;
 	}
 
-	if (pfile) {
+	if (options.username == NULL) {
+		options.username = strdup(getenv("USER"));
+	}
+
+	if (options.pwprompt) {
+		prompt_password();
+	}
```

Comment:
I was thinking of this logic, where if the option to prompt is selected, we just use that no matter what. Rather than falling back if it's not found in any places (`netrc`, `gpg`, `libsecret`).